immer/detail/hamts/bits.hpp

```diff
@@ -96,6 +107,16 @@ inline count_t popcount(std::uint64_t x)
 #endif
 }
 
+inline count_t popcount(std::uint16_t x)
+{
+    return popcount(static_cast<std::uint32_t>(x));
+}
+
+inline count_t popcount(std::uint8_t x)
+{
+    return popcount(static_cast<std::uint32_t>(x));
```

Comment:
I am not sure about this. The compilers provide optimized built-ins, and the the fallback can also be significantly optimized for lower sizes (at least for 16 bits). Always using the 32 bits version looks like a pessimization. Do you think they could be used?

Reply:
There is 16 bit popcount for msvc++, but I can't find gcc alternative. I decided to keep it simple for now.